File: content-solutions/sorting/raft-pairs.java
import java.util.*;

class Solution {
    public int solution(int[] weights, int limit) {
        int[] w = weights.clone();
        Arrays.sort(w);
        int i = 0, j = w.length - 1, rafts = 0;
        while (i <= j) {
            if (w[i] + w[j] <= limit) i++;
            j--;
            rafts++;
        }
        return rafts;
    }
}

File: content-solutions/sorting/raft-pairs.js
function solution(weights, limit) {
  const w = [...weights].sort((a, b) => a - b);
  let i = 0;
  let j = w.length - 1;
  let rafts = 0;
  while (i <= j) {
    if (w[i] + w[j] <= limit) i++;
    j--;
    rafts++;
  }
  return rafts;
}
